src/map_command_handler.hpp

```diff
@@ -190,9 +190,30 @@ class map_command_handler
 		}
 		else if (help_on_unknown_) {
 			utils::string_map symbols;
+		    // Get the input command on a string
+            std::string string_user_ = get_cmd();
+            // Get all commands (even if disallowed)
+            std::vector<std::string> command_list_ = get_commands_list();
+            // Initialize the distance and the command_proposal symbol so it
+            // does not appear if no candidate is found.
+            symbols["command_proposal"] = "";
+            int distance_ = 0;
+            // Compare the input with every command (excluding alias)
+            for (typename command_map::value_type i : command_map_) {
+                // No need to test commands that are not enabled
+                if(is_enabled(i.second)) {
+                    distance_ = approximate_string_distance(string_user_, i.first);
+                    // Maximum of a third of the letters are wrong
+                    if (distance_*100/std::min(string_user_.length(),i.first.length()) < 34){
+                        symbols["command_proposal"] = " did you mean '" + i.first + "'?";
+                        // If a good enough candidate is found, exit the loop.
+                        break;
+                    }
+                }
+            }
 			symbols["command"] = get_cmd();
 			symbols["help_command"] = cmd_prefix_ + "help";
-			print("help", VGETTEXT("Unknown command '$command', try $help_command "
+			print("help", VGETTEXT("Unknown command '$command',$command_proposal try $help_command "
```

Comment:
This looks translator-unfriendly. I'd recommend using an entirely separate string if there's a proposal, something like:

```suggestion
			if(has_command_proposal) {
				print("help", VGETTEXT("Unknown command '$command', did you mean '$command_proposal'? Or try $help_command "
			} else {
				print("help", VGETTEXT("Unknown command '$command', try $help_command "
			}
```

Where `has_command_proposal` would be true iff the `symbols["command_proposal"]` is populated with a command. Of course with this version `symbols["command_proposal"]` would be just the command alone, without that "did you mean" prefix.

Reply:
Can do. It was one of my concerns, translation issues, as I have no idea how translation works.